### PR TITLE
Example and Readme

### LINKED ISSUE #6: Add README
- what is this?
- how is it used, basically?
- does this require lit?

### LINKED ISSUE #7: Add example / test
Perhaps like:

https://codesandbox.io/s/xstate-react-typescript-template-wjdvn

where the button and text on toggle times are custom elements.

### PR DESCRIPTION
**Pull request by PaulHax**
closes #7 #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+This Lit ReactiveController:
+
+1. Subscribes to all updates of an XState actor or service.
+1. Runs the user provided state selector function when the state changes or the actor receives an event.
+1. Triggers the Lit component to update if the compare function returns false. The default compare function is `fast-deep-equal`
+
+## SelectState
+
+A wrapper for SelectController that pull the XState actor or service from a @lit-labs/context. This avoids needed to pass a shared actor down the component tree.
```
Comment (thewtex) [inline README.md]: pull -> pulls

**Review by thewtex — COMMENTED** [review]
@PaulHax thank you!

I get build errors:

```
> tsc && vite build

src/my-element.ts:3:50 - error TS2307: Cannot find module 'xstate' or its corresponding type declarations.

3 import { assign, createMachine, interpret } from 'xstate';
                                                   ~~~~~~~~

src/my-element.ts:23:31 - error TS7006: Parameter 'ctx' implicitly has an 'any' type.

23       entry: assign({ count: (ctx) => ctx.count + 1 }),
                                 ~~~

src/my-element.ts:34:58 - error TS18046: 'state' is of type 'unknown'.

34   count = new SelectorController(this, actor, (state) => state.context.count);
                                                            ~~~~~

../src/select-controller.ts:1:60 - error TS2307: Cannot find module 'lit' or its corresponding type declarations.

1 import { ReactiveController, ReactiveControllerHost } from 'lit';
                                                             ~~~~~

../src/select-controller.ts:2:54 - error TS2307: Cannot find module 'xstate' or its corresponding type declarations.

2 import { ActorRef, Subscribable, Subscription } from 'xstate';
                                                       ~~~~~~~~

../src/select-controller.ts:31:30 - error TS2345: Argument of type 'TEmitted | undefined' is not assignable to parameter of type 'TEmitted'.
  'TEmitted' could be instantiated with an arbitrary type which could be unrelated to 'TEmitted | undefined'.

31     this.selected = selector(getSnapshot(actorRef));
                                ~~~~~~~~~~~~~~~~~~~~~

../src/select-controller.ts:33:45 - error TS7006: Parameter 'emitted' implicitly has an 'any' type.

33     this.subscription = actorRef.subscribe((emitted) => {
                                               ~~~~~~~


Found 7 errors in 2 files.
```

**Comment by PaulHax** [comment]
Fixed the example build errors and spelling.  Made `fast-deep-equal` optional.

**Comment by thewtex** [comment]
@PaulHax thanks!

Builds now :1st_place_medal: 

New issue: I need to click twice on the button for the displayed count to increase.

**Comment by PaulHax** [comment]
Fixed the counter.

**Review by thewtex — APPROVED** [review]
@PaulHax thanks!!

**Pull request merged**